### PR TITLE
factorio: update all versions to 1.0.0

### PR DESCRIPTION
**Pull request by priegger**
###### Motivation for this change

factorio 1.0.0 was released: https://factorio.com/blog/post/fff-360

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/factorio/default.nix [hunk 1]
```diff
@@ -62,15 +62,15 @@ let
...
-        experimental  = bdist { sha256 = "0d64zzvp6zwz6p2izhhj998b6z8wd6r1b5p8mz1sbpz3v91sazj7"; version = "0.18.36"; };
+        stable        = bdist { sha256 = "0r0lplns8nxna2viv8qyx9mp4cckdvx6k20w2g2fwnj3jjmf3nc1"; version = "1.0.0"; };
+        experimental  = bdist { sha256 = "0r0lplns8nxna2viv8qyx9mp4cckdvx6k20w2g2fwnj3jjmf3nc1"; version = "1.0.0"; };
       };
       demo = {
-        stable        = bdist { sha256 = "07qknasaqvzl9vy1fglm7xmdi7ynhmslrb0a209fhbfs0s7qqlgi"; version = "0.17.79"; };
+        stable        = bdist { sha256 = "0h9cqbp143w47zcl4qg4skns4cngq0k40s5jwbk0wi5asjz8whqn"; version = "1.0.0"; };
       };
     };
     i686-linux = let bdist = bdistForArch { inUrl = "linux32"; inTar = "i386"; }; in {
```
Comment (prusnak) [inline pkgs/games/factorio/default.nix:76]: You missed the update of `i686-linux`

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/games/factorio/default.nix (hunk 1 above)
Comment (priegger) [reply pkgs/games/factorio/default.nix:76]: You are right, the commit comment suggests that all versions are updated to 1.0.0 but 0.14.23 is the latest 32bit version, see https://factorio.com/download/archive.

I will adjust the commit message.

**Comment by jonringer** [comment]
that's awkward, I was already doing then, and just wanted to make sure I wasn't duplicating PR before hitting submit. But you got the same hashes I did. So... looks good to me

**Review by jonringer — APPROVED** [review]
```
https://github.com/NixOS/nixpkgs/pull/95655
3 packages built:
factorio factorio-demo factorio-headless
```

**Pull request merged**